src/main/java/org/damiane/service/impl/TraineeServiceImpl.java

```diff
@@ -286,4 +287,71 @@ public List<TrainingDTO> getTraineeTrainingsList(String username, String passwor
         return getTraineeTrainingsHelper.mapToTrainingDTO(trainings);
     }
 
+
+
+    @Override
+    public List<TrainerResponse> updateTraineeTrainerList(String traineeUsername, List<String> trainerUsernames) {
+
+        log.info("Updating trainee's trainer list for trainee: {}", traineeUsername);
+
+        // Step 1: Find Trainee by Username
+        Trainee trainee = traineeRepository.findByUserUsername(traineeUsername);
+
+        if (trainee == null) {
+            log.info("Trainee not found with username: {}", traineeUsername);
+
```

Comment:
It's better to avoid returning null, because It can leed to NPE exceptions..
Try throwing your custom exception instead

Reply:
@NavrozashviliDamiane